### PR TITLE
Meta: Implemented formatter for MetaData language.

### PR DESCRIPTION
**Pull request by dalu2104**
Implemented a formatter for the MetaData lanuage in `elk.core.meta`. Needed artifacts were created accordingly.

Issue: #353

Signed-off-by: Daniel Lucas <stu124145@informatik.uni-kiel.de>

### CODE REVIEW

File: plugins/org.eclipse.elk.core.meta/src/org/eclipse/elk/core/meta/GenerateMetaData.mwe2 [hunk 1]
```diff
@@ -49,6 +49,9 @@ Workflow {
 			name = "org.eclipse.elk.core.meta.MetaData"
 			fileExtensions = "melk"
             
+            formatter =  {
```
Comment (uruuru) [inline plugins/org.eclipse.elk.core.meta/src/org/eclipse/elk/core/meta/GenerateMetaData.mwe2]: Indentation seems to be off.

File: plugins/org.eclipse.elk.core.meta/src/org/eclipse/elk/core/meta/formatting2/MetaDataFormatter.xtend [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+class MetaDataFormatter extends AbstractFormatter2 {
+
+    static val Procedure1<? super IHiddenRegionFormatter> one_space = [oneSpace]
+    static val Procedure1<? super IHiddenRegionFormatter> new_line = [newLine]
+    static val Procedure1<? super IHiddenRegionFormatter> indention = [indent]
+
+    @Inject extension MetaDataGrammarAccess
+    
+    def dispatch void format (MdBundleMember bundleMember, extension IFormattableDocument document){      
+        if(bundleMember instanceof MdAlgorithm){
```
Comment (uruuru) [inline plugins/org.eclipse.elk.core.meta/src/org/eclipse/elk/core/meta/formatting2/MetaDataFormatter.xtend]: Check your formatting of xtend files please. There should be a space after `if` and one before `{` and so on.

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk [hunk 3]
```diff
@@ -30,24 +30,23 @@ import org.eclipse.elk.core.util.RandomLayoutProvider
...
+    label "ELK"  
+    metadataClass core.options.CoreOptions 
 	idPrefix org.eclipse.elk 
 }
 
-category layered {
-	label "Layered"
-	description
+category layered {	
+    label "Layered"	description
```
Comment (uruuru) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk]: This looks odd, there should be newline before description, right?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk [hunk 4]
```diff
@@ -30,24 +30,23 @@ import org.eclipse.elk.core.util.RandomLayoutProvider
...
 		The nodes are arranged in layers, which are sometimes called \"hierarchies\", and then
 		reordered such that the number of edge crossings is minimized. Afterwards, concrete
 		coordinates are computed for the nodes and edge bend points."
 }
 
-category orthogonal {
-	label "Orthogonal"
+category orthogonal {	
+    label "Orthogonal"
 	description
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk]: This looks odd. Shouldn't `description` have the same indent as `label`?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk (hunk 3 above)
Comment (le-cds) [reply plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk]: Yep.

**Comment by dalu2104** [comment]
I formatted `Core.melk` with the implemented formatter in the latest commit, so the formatting rules should be visible there.

Similar to #383, my first commit to this branch was done with a wrong email adress, which is the reason the Eclipse validation failed. Hence, I had to rebase and amend said commit, which results in a force push.

**Pull request merged**